### PR TITLE
Deprecate AE2 recipe system.

### PR DESCRIPTION
**Pull request by fscan**
It's going away soon, replaced by custom JSON recipes and CraftTweaker support.

### CODE REVIEW

File: src/api/java/appeng/api/recipes/IIngredient.java [hunk 1]
```diff
@@ -29,7 +29,7 @@
 import appeng.api.exceptions.MissingIngredientException;
 import appeng.api.exceptions.RegistrationException;
 
-
+@Deprecated
 public interface IIngredient
```
Comment (orod-org) [inline src/api/java/appeng/api/recipes/IIngredient.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

File: src/api/java/appeng/api/recipes/ICraftHandler.java [hunk 2]
```diff
@@ -30,7 +30,7 @@
 import appeng.api.exceptions.RecipeException;
 import appeng.api.exceptions.RegistrationException;
 
-
+@Deprecated
 public interface ICraftHandler
```
Comment (orod-org) [inline src/api/java/appeng/api/recipes/ICraftHandler.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

File: src/api/java/appeng/api/features/IRecipeHandlerRegistry.java [hunk 3]
```diff
@@ -48,6 +48,7 @@
 	 * @param name name of crafthandler
 	 * @param handler class of crafthandler
 	 */
+	@Deprecated
 	void addNewCraftHandler( String name, Class<? extends ICraftHandler> handler );
```
Comment (orod-org) [inline src/api/java/appeng/api/features/IRecipeHandlerRegistry.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

File: src/api/java/appeng/api/features/IRecipeHandlerRegistry.java [hunk 4]
```diff
@@ -65,11 +66,13 @@
 	 * @return A recipe handler by name, returns null on failure.
 	 */
 	@Nullable
+	@Deprecated
 	ICraftHandler getCraftHandlerFor( String name );
```
Comment (orod-org) [inline src/api/java/appeng/api/features/IRecipeHandlerRegistry.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

File: src/api/java/appeng/api/features/IRecipeHandlerRegistry.java [hunk 5]
```diff
@@ -65,11 +66,13 @@
...
 
 	/**
 	 * @return a new recipe handler, which can be used to parse, and read recipe files.
 	 */
+	@Deprecated
 	IRecipeHandler createNewRecipehandler();
```
Comment (orod-org) [inline src/api/java/appeng/api/features/IRecipeHandlerRegistry.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

File: src/api/java/appeng/api/recipes/IRecipeLoader.java [hunk 6]
```diff
@@ -28,7 +28,7 @@
 
 import javax.annotation.Nonnull;
 
-
+@Deprecated
 public interface IRecipeLoader
```
Comment (orod-org) [inline src/api/java/appeng/api/recipes/IRecipeLoader.java]: CRITICAL Annotate the "IRecipeLoader" interface with the @FunctionalInterface annotation [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1609)
MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

File: src/api/java/appeng/api/recipes/IRecipeHandler.java [hunk 7]
```diff
@@ -27,6 +27,7 @@
 /**
  * Represents the AE2 Recipe Loading/Reading Class
  */
+@Deprecated
 public interface IRecipeHandler
```
Comment (orod-org) [inline src/api/java/appeng/api/recipes/IRecipeHandler.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)

**Comment by orod-org** [comment]
SonarQube analysis reported 20 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 12 major
* ![INFO][INFO] 7 info

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ICraftHandler.java#L53](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1d5d75cf80236ffb39a2b582bc7a4b1f4109c6aa/src/api/java/appeng/api/recipes/ICraftHandler.java#L53): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IIngredient.java#L45](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1d5d75cf80236ffb39a2b582bc7a4b1f4109c6aa/src/api/java/appeng/api/recipes/IIngredient.java#L45): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IIngredient.java#L56](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1d5d75cf80236ffb39a2b582bc7a4b1f4109c6aa/src/api/java/appeng/api/recipes/IIngredient.java#L56): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IIngredient.java#L91](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1d5d75cf80236ffb39a2b582bc7a4b1f4109c6aa/src/api/java/appeng/api/recipes/IIngredient.java#L91): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IRecipeLoader.java#L41](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1d5d75cf80236ffb39a2b582bc7a4b1f4109c6aa/src/api/java/appeng/api/recipes/IRecipeLoader.java#L41): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00112)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by yueh** [comment]
There should be a note about crafttweaker support being temporary due to resource packs are lacking a few features currently and our recipe system needing a major overhaul, which is already obsolete in most cases due to resource packs and the extensions provided by forge for it.

1.13 plans to introduce data packs, which should solve the shortcomings of resource packs and allow for different recipe per server. Should CT have issues with supporting it or cause more issues like the timeout problems due to reloading JEI it will be dropped in favour of a vanilla/forge solution.

**Comment by orod-org** [comment]
SonarQube analysis reported 20 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 12 major
* ![INFO][INFO] 7 info

Watch the comments in this conversation to review them.

#### 5 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ICraftHandler.java#L53](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/bc3064ea95d0e15bfd338b960a6eddc12336fe32/src/api/java/appeng/api/recipes/ICraftHandler.java#L53): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IIngredient.java#L45](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/bc3064ea95d0e15bfd338b960a6eddc12336fe32/src/api/java/appeng/api/recipes/IIngredient.java#L45): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IIngredient.java#L56](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/bc3064ea95d0e15bfd338b960a6eddc12336fe32/src/api/java/appeng/api/recipes/IIngredient.java#L56): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IIngredient.java#L91](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/bc3064ea95d0e15bfd338b960a6eddc12336fe32/src/api/java/appeng/api/recipes/IIngredient.java#L91): Refactor this method to throw at most one checked exception instead of: appeng.api.exceptions.RegistrationException, appeng.api.exceptions.MissingIngredientException [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1160)
1. ![MAJOR][MAJOR] [IRecipeLoader.java#L41](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/bc3064ea95d0e15bfd338b960a6eddc12336fe32/src/api/java/appeng/api/recipes/IRecipeLoader.java#L41): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00112)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**